webpack.config.js

```diff
@@ -90,16 +120,26 @@ module.exports = {
 		],
 	},
 	mode,
-	entry: gutenbergPackages.reduce( ( memo, packageName ) => {
-		const name = camelCaseDash( packageName );
-		memo[ name ] = `./packages/${ packageName }`;
-		return memo;
-	}, {} ),
+	entry: createEntrypoints(),
 	output: {
 		devtoolNamespace: 'wp',
-		filename: './build/[basename]/index.js',
+		filename: ( data ) => {
+			const { chunk } = data;
+			const { entryModule } = chunk;
+			const { rawRequest } = entryModule;
+
+			/*
+			 * If the file being built is a Core Block's frontend file,
+			 * we build it in the block's directory.
+			 */
+			if ( rawRequest && rawRequest.includes( '/frontend.js' ) ) {
+				return `./build/block-library/blocks/[name]/frontend.js`;
+			}
+
+			return './build/[name]/index.js';
+		},
 		path: __dirname,
-		library: [ 'wp', '[name]' ],
+		library: [ 'wp', '[camelName]' ],
```

Comment:
One more thing that needs more thinking. I don't think we want to produce `wp` globals for frontend scripts. Taking into account the comment from @pento https://github.com/WordPress/gutenberg/pull/30341/files#r614557004, it might be worth exploring whether to use two independent webpack configs to gain more flexibility in how those two different use cases are handled. We maintain multiple (2) webpack configs in WordPress core: 
https://github.com/WordPress/wordpress-develop/blob/40728234568b07d6bcaa456816bf284f8c5cc670/webpack.config.js#L13-L16